fullcalendar-app/src/components/FullCalendar.vue

```diff
@@ -0,0 +1,410 @@
+    },
+    addEvent(newEvent) {
+      this.addEventToCalendar(newEvent);
+      this.closePopup()
+    },
+    addEventToCalendar(newEvent) {
+      const calendarApi = this.$refs.fullCalendar.getApi();
+      newEvent.id = newEvent.nid + '-' + Math.random().toString(16).slice(2);
+      console.log(newEvent)
+
+      calendarApi.addEvent(newEvent);
+    },
+    async handleWeekChange(payload) {
+      // We check that initialization is complete before calling loadEvents.
+      if (this.initializationCompleted) {
+        const { startStr, endStr } = payload;
+        await this.loadEvents(startStr, endStr);
+        updateUrlParams(startStr, endStr);
+      }
+    },
+    handleCategoryChange(categories, isMounted) {
+      const calendarApi = this.$refs.fullCalendar.getApi();
+      const start = calendarApi.view.activeStart.toISOString();
+      const end = calendarApi.view.activeEnd.toISOString();
+
+      this.selectedCategories = categories
+      // If the function is called on mount, we avoid calling loadEvents again.
+      if (!isMounted) {
+        this.loadEvents(start, end);
+      }
+
+      updateUrlParams(start, end, categories);
+    },
+    async loadEvents(start, end, isInitialization = false) {
+      // A condition to prevent redundant calls during initialization.
+      if (isInitialization || this.initializationCompleted) {
+        this.eventService.getEvents(start, end, this.selectedCategories)
+          .then(data => {
+            this.calendarOptions.events = data;
+
+            // We establish that the initialization is complete.
+            if (isInitialization) this.initializationCompleted = true;
+          });
+      }
+    },
+    checkSameDaySelection(selectInfo) {
+      const start = selectInfo.start;
+      const end = selectInfo.end;
+      // Checking that the end of the selection is the same day (does not include the next day).
+      return start.toISOString().substring(0, 10) === end.toISOString().substring(0, 10);
+    },
+    updateEvent(eventInfo) {
+      const event = {
+        id: eventInfo.event.extendedProps.nid,
+        nid: eventInfo.event.extendedProps.nid,
+        startGlobal: combineDateTime(eventInfo.event.extendedProps.startGlobal, eventInfo.event.start),
+        endGlobal: combineDateTime(eventInfo.event.extendedProps.endGlobal, eventInfo.event.end),
+      };
+       this.eventService.updateEventOnServer(event);
+    },
+    getUrlParams() {
+      const urlParams = new URLSearchParams(window.location.search);
+
+      return {
+        start: urlParams.get('start'),
+        end: urlParams.get('end'),
+        categories: urlParams.get('categories')?.split(',') || []
+      };
+    },
+    handleEventClick(clickInfo) {
+      // Approximate width of the popover.
+      const popoverWidth = 300;
+      // Approximate height of the popover.
+      const popoverHeight = 200;
+      // Minimum space between the popover and the edge of the window.
+      const windowPadding = 10;
```

Comment:
drupalSettings from config?

Reply:
This method was completely rewritten, this hardcoded data is not needed, we need to determine the width and height of the popover dynamically and dynamically place it on the page. Therefore, this data was deleted.